rcl/src/rcl/parameter.c

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <rosidl_generator_c/string_functions.h>
+#include <rcl_interfaces/msg/parameter__struct.h>
+#include <rcl_interfaces/msg/parameter_event__struct.h>
+#include <rcl_interfaces/msg/parameter_type__functions.h>
+
+#include "rcl/types.h"
+
+// TODO: Can we assume that other ParameterValue fields are set to NOT_ASSIGNED or should we reassign them?
```

Comment:
I guess this is a question about the "default" constructor of the ParameterValue's generated structure and what it sets its values to initially? I suppose you should be able to specify the default values in the message definition if the implicit behavior is not correct.